lib/utils/display.js

```diff
@@ -161,43 +154,54 @@ class Display {
     }
   }
 
-  load ({
+  get chalk () {
+    return {
+      noColor: this.#noColorChalk,
+      stdout: this.#stdoutChalk,
+      stderr: this.#stderrChalk,
+    }
+  }
+
+  async load ({
     heading,
     json,
     loglevel,
     progress,
-    stderrChalk,
     stderrColor,
-    stdoutChalk,
     stdoutColor,
     timing,
     unicode,
   }) {
+    // get createSupportsColor from chalk directly if this lands
+    // https://github.com/chalk/chalk/pull/600
+    const [{ Chalk }, { createSupportsColor }] = await Promise.all([
+      import('chalk'),
+      import('supports-color'),
+    ])
+    // we get the chalk level based on a null stream meaning chalk will only use
+    // what it knows about the environment to get color support since we already
+    // determined in our definitions that we want to show colors.
+    const level = Math.max(createSupportsColor(null).level, 1)
+
+    this.#noColorChalk = new Chalk({ level: 0 })
```

Comment:
What are we optimizing here for?  If they use color then this chalk instance didn't need to be created.  Let's look at the matrix of possibilities here and see if this is more work than it's worth